### PR TITLE
blast single column layout for smaller screen

### PR DESCRIPTION
**Pull request by ens-ma7**
## Description
Improving the single column layout for blast when selecting between sequence and species for smaller screen. This is now split into two buttons instead of one button with different text.

## Related JIRA Issue(s)
https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-1572

## Deployment URL(s)
http://blast-single-column.review.ensembl.org

## Views affected
blast: http://blast-single-column.review.ensembl.org/blast

### CODE REVIEW

File: src/content/app/tools/blast/components/blast-input-sequences/BlastInputSequences.scss [hunk 1]
```diff
@@ -18,6 +18,14 @@
   background-color: $white;
 }
 
+.smallScreenHeader {
+  top: 49px;
+}
+
+.mainContainerSmall .header {
+  top: 49px;
+}
```
Comment (azangru) [inline src/content/app/tools/blast/components/blast-input-sequences/BlastInputSequences.scss:27]: Where is this used?

File: src/content/app/tools/blast/components/blast-input-sequences/BlastInputSequences.scss [hunk 2]
```diff
@@ -18,6 +18,14 @@
   background-color: $white;
 }
 
+.smallScreenHeader {
+  top: 49px;
+}
```
Comment (azangru) [inline src/content/app/tools/blast/components/blast-input-sequences/BlastInputSequences.scss:23]: What's this for?

https://user-images.githubusercontent.com/6834224/167392653-f0bdb01f-2f68-42b4-acb7-d28b5a7cbd90.mp4

`top` wouldn't work on `position: static` (which is the default for a DOM element)

File: src/content/app/tools/blast/components/blast-species-selector/BlastSpeciesSelector.scss [hunk 3]
```diff
@@ -13,6 +13,10 @@
   background-color: $white;
 }
 
+.smallScreenHeader {
+  top: 49px;
+}
```
Comment (azangru) [inline src/content/app/tools/blast/components/blast-species-selector/BlastSpeciesSelector.scss:18]: Same question as above

File: src/content/app/tools/blast/components/blast-species-selector/BlastSpeciesSelectorHeader.tsx [hunk 4]
```diff
@@ -37,43 +33,35 @@ const BlastSpeciesSelectorHeader = (props: Props) => {
...
       </div>
-      <div className={styles.headerGroup}>
-        {!compact && (
+      {compact && (
+        <span className={styles.clearAll} onClick={onClearAll}>
+          Clear all
+        </span>
+      )}
+      {!compact && (
+        <div className={styles.headerGroup}>
```
Comment (azangru) [inline src/content/app/tools/blast/components/blast-species-selector/BlastSpeciesSelectorHeader.tsx:59]: what's the purpose of this wrapper for a single "Clear all" element?

File: src/content/app/tools/blast/views/blast-form/BlastForm.scss [hunk 5]
```diff
@@ -31,3 +31,20 @@
   display: flex;
   flex-direction: column;
 }
+
+.sequenceSpeciesSelector {
```
Comment (azangru) [inline src/content/app/tools/blast/views/blast-form/BlastForm.scss:35]: `blastFormStepToggle`? (we have a notion of "steps" for the single-column form)

File: src/content/app/tools/blast/views/blast-form/BlastForm.scss [hunk 6]
```diff
@@ -31,3 +31,20 @@
...
+  gap: 10px;
+  max-width: 1660px;
+  padding: 0 20px 20px 0;
+  background-color: white;
+  position: sticky;
+  top: 0px;
+  z-index: 1;
+}
+
+.selectedSequenceSpeciesButton {
```
Comment (azangru) [inline src/content/app/tools/blast/views/blast-form/BlastForm.scss:47]: just `.buttonActive`?

File: src/content/app/tools/blast/views/blast-form/BlastForm.tsx [hunk 7]
```diff
@@ -106,4 +115,41 @@ const MainSmall = () => {
   );
 };
 
+const BlastSequenceSpeciesSelector = () => {
```
Comment (azangru) [inline src/content/app/tools/blast/views/blast-form/BlastForm.tsx:118]: `BlastFormStepToggle`?

File: src/content/app/tools/blast/components/blast-species-selector/BlastSpeciesSelector.scss (hunk 3 above)
Comment (ens-ma7) [reply src/content/app/tools/blast/components/blast-species-selector/BlastSpeciesSelector.scss:18]: @azangru  This and all the above top margin are related with the position sticky when you scroll the page. The button should be fixed at the top when you scroll the page.

File: src/content/app/tools/blast/components/blast-species-selector/BlastSpeciesSelectorHeader.tsx (hunk 4 above)
Comment (ens-ma7) [reply src/content/app/tools/blast/components/blast-species-selector/BlastSpeciesSelectorHeader.tsx]: My bad, i missed cleaning this one up and changing things around. I have pushed the changes now.

File: src/content/app/tools/blast/views/blast-form/BlastForm.tsx [hunk 8]
```diff
@@ -15,12 +15,19 @@
...
 
 import React, { useEffect, useRef } from 'react';
-import { useSelector } from 'react-redux';
+import { useSelector, useDispatch } from 'react-redux';
+import classNames from 'classnames';
 
 import { useBlastConfigQuery } from 'src/content/app/tools/blast/state/blast-api/blastApiSlice';
 import useMediaQuery from 'src/shared/hooks/useMediaQuery';
 
+import { SecondaryButton } from 'src/shared/components/button/Button';
```
Comment (imransl) [inline src/content/app/tools/blast/views/blast-form/BlastForm.tsx:24]: Group this component import with the tools component imports?

**Comment by jyothishnt** [comment]
Hi Ridwan, I see the layout changing at 1900px, and the buttons appeared didnt align with the Job List button. Is this expected?

**Comment by ens-ma7** [comment]
@jyothishnt fix the width layout, can you please check again and let me know if you spot anything else.

@imransl i also made the changes you suggested, can you please review. thanks

**Comment by azangru** [comment]
Is there a hint of border-left I am seeing here?

https://user-images.githubusercontent.com/6834224/168276752-c5811038-81df-4ffa-9ed9-caa7664d00d5.mp4

File: src/content/app/tools/blast/components/blast-species-selector/BlastSpeciesSelector.scss (hunk 3 above)
Comment (azangru) [reply src/content/app/tools/blast/components/blast-species-selector/BlastSpeciesSelector.scss:18]: Thanks, but I am still confused. `.mainContainerSmall .header` and `.smallScreenHeader` seem to target the same DOM element; why the duplication of the CSS rule?

image

File: src/content/app/tools/blast/views/blast-form/BlastForm.scss [hunk 9]
```diff
@@ -31,3 +31,20 @@
   display: flex;
   flex-direction: column;
 }
+
+.blastFormStepToggle {
+  display: flex;
+  justify-content: flex-end;
+  gap: 10px;
+  max-width: 1790px;
```
Comment (azangru) [inline src/content/app/tools/blast/views/blast-form/BlastForm.scss:39]: Is this a magic number related to other magic numbers on the BLAST pages? If yes, could you leave a TODO comment saying that we need to group those magic numbers into variables? See all the TODOs in the merged PR https://github.com/Ensembl/ensembl-client/pull/745

There is https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-1602 which would make a very nice addition to sprint 9. It's about cleaning up the code. It doesn't specifically talk about all these shared widths on BLAST pages. We probably need a dedicated ticket for that; would also make a good addition to sprint 9.

**Comment by imransl** [comment]
> Is there a hint of border-left I am seeing here?

Can't reproduce this on my machine. How can a border be seen when the background of the header is not transparent 🤔

File: src/content/app/tools/blast/components/blast-species-selector/BlastSpeciesSelector.scss (hunk 3 above)
Comment (ens-ma7) [reply src/content/app/tools/blast/components/blast-species-selector/BlastSpeciesSelector.scss:19]: @azangru you are right it is an extra rule that we dont need, i have removed it.

**Comment by ens-ma7** [comment]
> Is there a hint of border-left I am seeing here?
> 
>  header.mp4

I have fixed this now, had to play with the margin-left and width.

**Comment by jyothishnt** [comment]
Hi @ens-ma7 I see some alignment issues for the button at 1900px as below.


Further shrinking of window leads to this.

**Comment by ens-ma7** [comment]
Thanks for spotting this @jyothishnt i thought it was fixed but I broke it when i was fixing something else. Can you please review again.

**Review by azangru — APPROVED** [review]

File: src/content/app/tools/blast/views/blast-form/BlastForm.scss [hunk 10]
```diff
@@ -31,3 +33,21 @@
...
 }
+
+.blastFormStepToggle {
+  display: flex;
+  justify-content: flex-end;
+  gap: 10px;
+  max-width: 1770px; //TODO: To replace with global variables for spaces on the blast form (see ENSWBSITES-1602)
+  margin-left: -2px;
+  padding: 20px 0 20px 0;
+  background-color: white;
```
Comment (imransl) [inline src/content/app/tools/blast/views/blast-form/BlastForm.scss:44]: Use `$white` instead of `white`?

**Review by imransl — APPROVED** [review]

**Review by jyothishnt — APPROVED** [review]
Not fully satisfied, but its ok :-)

**Comment by azangru** [comment]
I don't understand... I thought the whole point of the exercise with `position: relative` and `z-index` was to get rid of margin-bottom of the grey `topBar`, and to move the responsibility for the white space to the main container. But the margin-bottom is still there. What was it that you were trying to achieve then?

image

**Comment by ens-ma7** [comment]
@azangru Confirm the space between the topbar and maincontainer is 20px; Dev spec if you want to have a look at https://xd.adobe.com/view/4bd9ad02-5287-4868-9cfe-1bd59f362519-1e8a/

I have removed the extra margin top and merged dev to the branch. I am not sure if it will be fine to just merge this branch (the commit merge branch to 'dev' wont cause any problem?)

**Review by azangru — APPROVED** [review]

**Pull request merged**